### PR TITLE
RND203:Add immediately sending write commands

### PR DESCRIPTION
**Pull request by nzulinski-mobica**

### CODE REVIEW

File: sys/dev/ath/if_ath_usb_def.h [hunk 1]
```diff
@@ -525,6 +525,7 @@ struct ath_usb_softc {
 	void				(*sc_node_free)(struct ieee80211com *,
 					    struct ieee80211_node *);
 	int				sc_key_tasks;
+	bool				interrupt;
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/if_ath_usb_def.h]: Not sure if `interrupt` name for this property is correct. Maybe `immediate_write`, `no_buffer_write` etc. would be better?

Comment (nzulinski-mobica) [reply sys/dev/ath/if_ath_usb_def.h]: I changed to no_buffer_write

File: sys/dev/ath/if_ath_usb.c [hunk 2]
```diff
@@ -1673,7 +1675,9 @@ ath_usb_write(struct ath_softc *sc, uint32_t addr, uint32_t val)
 
 	usc->wbuf[usc->wcount].addr = htobe32(addr);
 	usc->wbuf[usc->wcount].val  = htobe32(val);
-	if (++usc->wcount == AR_MAX_WRITE_COUNT)
+	if (!usc->no_buffer_write & (++usc->wcount == AR_MAX_WRITE_COUNT))
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/if_ath_usb.c]: Shouldn't it be more like (without `else`):
```C
	if ((++usc->wcount == AR_MAX_WRITE_COUNT) || (usc->no_buffer_write))
		ath_usb_write_barrier(sc);
```

Comment (nzulinski-mobica) [reply sys/dev/ath/if_ath_usb.c]: I don't think so because then we inceremnt usc->wcount when no_buffer_write is true

Comment (nzulinski-mobica) [reply sys/dev/ath/if_ath_usb.c]: I changed as you suggested.

**Review by Mikolaj-Filar-Mobica — APPROVED** [review]

**Review by Rafal-Litwinski-Mobica — APPROVED** [review]

**Pull request merged**